### PR TITLE
feat(web) #74 Favoring projects

### PR DESCRIPTION
**Pull request by galaplexus**
Using localstorage
Most basic UI differentiation between favorite projects and non favorites.
Most direct implementation. Definitely not the best algorithms, but does the trick and nobody will manipulate soo many projects.

### CODE REVIEW

File: web/src/components/Project.vue [hunk 1]
```diff
@@ -1,7 +1,8 @@
 <template>
-  <router-link :to="`/${project}/${latestVersion}`">
-    <md-card>
-      <md-card-header>
+  
+  <md-card>
+      <router-link :to="`/${project}/${latestVersion}`">
+    <md-card-header>
```
Comment (randombenj) [inline web/src/components/Project.vue:5]: Some strange reformatting going on?

File: web/src/components/Project.vue [hunk 2]
```diff
@@ -23,26 +30,39 @@ import ProjectRepository from '@/repositories/ProjectRepository'
...
   async created() {
     document.title = this.project + " | docat"
     this.logoURL = ProjectRepository.getProjectLogoURL(this.project)
     this.versions = await ProjectRepository.getVersions(this.project)
     this.latestVersion = (this.versions.find((version) => version.name == 'latest') || this.versions[0]).name
+    this.isFavourite = ProjectRepository.isFavourite(this.project)
+  },
+  methods: {
+    switchFavourite() {
+      ProjectRepository.setFavourite(this.project, !this.isFavourite);
```
Comment (randombenj) [inline web/src/components/Project.vue:53]: I think we decided to not use semicolons (don't know if this was a good idea) but I would like to keep it consistant :)

**Comment by randombenj** [comment]
Cool stuff! I think local storage is a great way to do this :tada:

File: web/src/components/Project.vue [hunk 3]
```diff
@@ -23,26 +30,39 @@ import ProjectRepository from '@/repositories/ProjectRepository'
...
+    }
   }
 }
 </script>
 <style lang="scss">
 .md-avatar.hidden {
   display: none;
 }
+#favourite-star {
+  color: yellow;
```
Comment (randombenj) [inline web/src/components/Project.vue:64]: I would go for `color: #fff;` and change the logic (bordered means not favourited, filled means favourited)

Comment (galaplexus) [reply web/src/components/Project.vue:64]: Not sure I understand you here
You want yellow with #ffff00 or white with #fff ?
You would have a border only for not-favorites?

Comment (randombenj) [reply web/src/components/Project.vue:64]: Ah sorry I mean white #fff or #ffffff :slightly_smiling_face: 
Yes, so only border for non-favourites and fully filled out for favourites.

**Comment by fliiiix** [comment]
I finally had time to review this and made a few fixes myself already what still needs to happen

Todo
- [ ] on some screen sizes the md-card gets huge
image
- [ ] create 1 clean commit (can be done at the very end)

Optional
- [ ] live update if i favor a project it only shows up in favorites after a page refresh
- [ ] If i don't have any favorites we waist quite a bit of space 
image
- [ ] Placement of the star is off-center
image

**Comment by galaplexus** [comment]
Thanks for feedback.
Not sure what you mean with star off-center.

I believe I improved the responsiveness. It would be very hard to do better now.
image

Once it's good I'll make a nice single commit

**Comment by fliiiix** [comment]
breaks CI in the current state so there is that

**Comment by randombenj** [comment]
@galaplexus @fliiiix I resolved the conflicts, changes look good for me, @galaplexus thanks for the nice change!
I think you need to squash the commits, as I will loose permission to push to your branch once it's a single commit.

**Comment by fliiiix** [comment]
Sounds good thanks @randombenj :sparkles: 

@galaplexus please create a good commit then i will review everything and get it merged

**Comment by randombenj** [comment]
Thanks a lot for this awesome change @galaplexus ! :tada: :partying_face:

**Pull request merged**